include/matx/transforms/fft/fft_fftw.h

```diff
@@ -0,0 +1,545 @@
+                                      reinterpret_cast<fftwf_complex*>(out_ptr), 
+                                      params.onembed, 
+                                      params.ostride, 
+                                      params.odist,  
+                                      FFTW_ESTIMATE);
+        }
+
+        fftwf_execute(plan);
+        fftwf_destroy_plan(plan);
+        fftwf_cleanup();         
+      }
+      else {
+        if constexpr (DeduceFFTTransformType<OutputTensor, InputTensor>() == FFTType::Z2Z) {
+          plan  = fftw_plan_many_dft( 1, 
+                                      params.n, 
+                                      params.batch, 
+                                      reinterpret_cast<fftw_complex*>(in_ptr), 
+                                      params.inembed, 
+                                      params.istride, 
+                                      params.idist,
+                                      reinterpret_cast<fftw_complex*>(out_ptr), 
+                                      params.onembed, 
+                                      params.ostride, 
+                                      params.odist,  
+                                      fft_dir, 
+                                      FFTW_ESTIMATE);
+        }
+        else if constexpr (DeduceFFTTransformType<OutputTensor, InputTensor>() == FFTType::Z2D) {
+          plan  = fftw_plan_many_dft_c2r( 1, 
+                                      params.n, 
+                                      params.batch, 
+                                      reinterpret_cast<fftw_complex*>(in_ptr), 
+                                      params.inembed, 
+                                      params.istride, 
+                                      params.idist,
+                                      out_ptr, 
+                                      params.onembed, 
+                                      params.ostride, 
+                                      params.odist,  
+                                      FFTW_ESTIMATE);
+        }
+        else if constexpr (DeduceFFTTransformType<OutputTensor, InputTensor>() == FFTType::D2Z) {
+          plan  = fftw_plan_many_dft_r2c( 1, 
+                                      params.n, 
+                                      params.batch, 
+                                      in_ptr, 
+                                      params.inembed, 
+                                      params.istride, 
+                                      params.idist,
+                                      reinterpret_cast<fftw_complex*>(out_ptr), 
+                                      params.onembed, 
+                                      params.ostride, 
+                                      params.odist,  
+                                      FFTW_ESTIMATE);
+        } 
+        
+        fftw_execute(plan);
+        fftw_destroy_plan(plan);
+        fftw_cleanup();           
+      }
+
+      return plan;
```

Comment:
The plan has been destroyed above; do we want to return it? The return appears unused. Do we have caching for NVPL plans?

Reply:
Thanks. This was done prior to knowing nvpl fft's limitations to where having a plan stick around is not beneficial.

Reply:
should we return null instead then?

Reply:
yes, it's a void lambda now